### ISSUE TITLE
Extbase's htmlResponse() is added wrongly

### ISSUE THREAD

**Issue by brotkrueml** [sabbelasichon/typo3-rector#2764]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 6861ae9a4fe8e52dd4633101d80c9c55c1093b9e (0.12.9)             |

ExtbaseControllerActionsMustReturnResponseInterfaceRector inserts wrong htmlResponse().

## Minimal PHP Code Causing Issue

Sorry, can't reproduce it on getrector.org/demo. But I have two issues in the mentioned rector rule in an Extbase controller:

### htmlReponse() is wrapped around ForwardResponse

    public function someAction(): ResponseInterface
    {
        return new \TYPO3\CMS\Extbase\Http\ForwardResponse('another');
    }

is changed to:

    public function someAction(): ResponseInterface
    {
        return $this->htmlResponse(new \TYPO3\CMS\Extbase\Http\ForwardResponse('another'));
    }

### htmlResponse() is added when PropagateResponseException is used

This is the case when providing a download of a file to circumvent page renderer.

     public function someAction(): ResponseInterface
    {
        $response = $this->responseFactory->createResponse(200);
        throw new \TYPO3\CMS\Core\Http\PropagateResponseException($response, 200);
    }

is changed to:

    public function someAction(): ResponseInterface
    {
        $response = $this->responseFactory->createResponse(200);
        throw new \TYPO3\CMS\Core\Http\PropagateResponseException($response, 200);
        return $this->htmlResponse();
    }

## Expected Behaviour

The code should be leaved untouched if a ResponseInterface is already returned. Also when a PropagateResponseException is thrown.

**Comment by julianhofmann** [comment]
ExtbaseControllerActionsMustReturnResponseInterfaceRector changes also abstract methods... 
(Not reproducablet on getrector.org/demo)

```
 -    public abstract function listAction(?Filter $filter = null, string $letter = ''): \Psr\Http\Message\ResponseInterface;
 +    public abstract function listAction(?Filter $filter = null, string $letter = ''): \Psr\Http\Message\ResponseInterface
 +    {
 +        return $this->htmlResponse();
 +    }
```

**Comment by sabbelasichon** [comment]
@julianhofmann: Sure, this is a bug. It is fixed by #2812
